starlingbank/__init__.py

```diff
@@ -122,8 +123,43 @@ def get_image(self, filename: str = None) -> None:
             file.write(b64decode(base64_image))
 
 
+"""Representation of a Spending Category."""
+class SpendingCategory:
+
+    def __init__(
+        self, auth_headers: Dict, sandbox: bool, account_uid: str
+    ) -> None:
+        self._auth_headers = auth_headers
+        self._sandbox = sandbox
+        self._account_uid = account_uid
+
+        self.month = None
+        self.year = None
+        self.spending_category = None
+        self.net_direction = None
+        self.currency = None
+        self.total_spent = 0.0
+        self.total_received = 0.0
+        self.net_spend = 0.0
+        self.percentage = 0.0
+        self.transaction_count = 0
+
+    def update(self, category: Dict = None, month: str = None, year: str = None) -> None:
```

Comment:
Unlike `SavingsGoal` objects, `SpendingCategory` objects won't be responsible for updating themselves so I think we can do away with an update method altogether.

Also see my note below on the `update_spending_categories_data` method.

Reply:
So is your suggestion to declare all the parameters that I've added to the update method in the __init__ method instead?